tests/Template/ConditionalReturnTypeTest.php

```diff
@@ -859,6 +859,22 @@ function test(string $s): void {
                     }
                     '
             ],
+            'returnTypeBasedOnPhpVersionId' => [
+                '<?php
+                    /**
+                     * @psalm-return (PHP_VERSION_ID is int<70300, max> ? string : int)
+                     */
+                     function getSomething()
+                     {
+                        return mt_rand(1, 10) > 5 ? "a value" : 42;
+                     }
+
+                     $something = getSomething();
+                ',
+                ['$something' => 'int'],
+                [],
+                '7.2'
```

Comment:
You may want to add another test with the same code to show up it returns `string` on PHP 7.3+

Reply:
Thanks, added another check to that test with essentially the same function but a range starting from PHP 7.1